plugins/scenario/hu.bme.mit.gamma.scenario.language/src/hu/bme/mit/gamma/scenario/language/linking/ScenarioLanguageLinker.xtend

```diff
@@ -0,0 +1,99 @@
+/********************************************************************************
+ * Copyright (c) 2020-2021 Contributors to the Gamma project
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * SPDX-License-Identifier: EPL-1.0
+ ********************************************************************************/
+package hu.bme.mit.gamma.scenario.language.linking
+
+import com.google.inject.Inject
+import hu.bme.mit.gamma.scenario.model.ScenarioDeclaration
+import hu.bme.mit.gamma.scenario.model.ScenarioModelPackage
+import org.eclipse.emf.common.util.URI
+import org.eclipse.emf.ecore.EObject
+import org.eclipse.emf.ecore.EReference
+import org.eclipse.emf.ecore.resource.impl.ResourceSetImpl
+import org.eclipse.xtext.conversion.IValueConverterService
+import org.eclipse.xtext.linking.impl.DefaultLinkingService
+import org.eclipse.xtext.nodemodel.INode
+
+class ScenarioLanguageLinker extends DefaultLinkingService {
```

Comment:
This entire class is code duplication now, which can be avoided by extending the `GammaLangaugeLinker` class. See `StatechartLanguageLinker` on how it should be done.